Rebus.AdoNet.Tests/SagaPersisterTests.cs

```diff
@@ -318,31 +399,127 @@ public void WhenIgnoringNullProperties_DoesNotSaveNullPropertiesOnInsert()
+			{
+				Assert.Multiple(() =>
+				{
+					var correlationsType = schema.First(x => x.Item1 == "correlations").Item2;
+					Assert.That(tableNames, Does.Not.Contains(SagaIndexTableName), "Have we created saga indexes table?!");
+					Assert.That(Dialect.GetDbTypeFor(correlationsType), Is.EqualTo(DbType.Object), "#2.0");
+					Assert.That(correlationsType, Is.EqualTo("jsonb"), "#2.1");
 
-			// assert
-			var existingTables = GetTableNames();
-			Assert.That(existingTables, Contains.Item(SagaIndexTableName));
-			Assert.That(existingTables, Contains.Item(SagaTableName));
+					if (Dialect.SupportsGinIndexes)
+					{
+						var ix = Dialect.SupportsMultiColumnGinIndexes ? $"ix_{SagaTableName}_saga_type_correlations" : $"ix_{SagaTableName}_correlations";
+						Assert.That(indexes, Contains.Item(ix), "#2.2");
+					}
+				});
+			}
+			else //< If someday we add another persister.. we should add coverage for it.
+			{
+				throw new NotSupportedException("Missing coverage for a new persister?!?!");
+			}
 		}
 
 		[Test]
 		public void DoesntDoAnythingIfTheTablesAreAlreadyThere()
 		{
-			// arrange
-			var persister = CreatePersister();
-			ExecuteCommand(@"CREATE TABLE """ + SagaTableName + @""" (""id"" INT NOT NULL)");
-			ExecuteCommand(@"CREATE TABLE """ + SagaIndexTableName + @""" (""id"" INT NOT NULL)");
+			Assert.That(() =>
+			{
+				_persister.EnsureTablesAreCreated();
+				_persister.EnsureTablesAreCreated();
+				_persister.EnsureTablesAreCreated();
+			}, Throws.Nothing);
+		}
 
-			// act
+		[Test]
+		public void SagaDataCanBeRecoveredWithDifferentKindOfValuesAsCorrelations()
+		{
+			var data = new ComplexSaga()
+			{
+				Id = Guid.NewGuid(),
+				Tuple = Tuple.Create("z", (object)129310.23D),
+				Uuid = Guid.NewGuid(),
+				Char = '\n',
+				Text = "this is a string w/ spanish characters like EÑE.",
+				Bool = true,
+				SByte = sbyte.MinValue,
+				Byte = byte.MaxValue,
+				UShort = ushort.MinValue,
+				Short = short.MaxValue,
+				UInt = uint.MinValue,
+				Int = int.MaxValue,
+				ULong = ulong.MinValue,
+				Long = long.MaxValue,
+				Float = float.NaN,
+				Double = double.PositiveInfinity,
+				Decimal = 10000.746525344148M,
+				Date = DateTime.UtcNow,
+				Time = TimeSpan.MaxValue,
+				Enum = ComplexSaga.Values.Valid,
+				EnumFlags = ComplexSaga.Flags.One | ComplexSaga.Flags.Two,
+				Object = new { one = 1, two = 2.7878798745M },
+				Strings = new[] { "x", "y" },
+				Decimals = new[] { 0.646584564M, 6.98984564544212M },
+				Objects = new object[] { 'x', 0x1, float.Epsilon },
+				Bag = new Dictionary<string, object>() { { "a", 1 }, { "b", float.NaN } }
+			};
 
-			// assert
-			persister.EnsureTablesAreCreated();
-			persister.EnsureTablesAreCreated();
-			persister.EnsureTablesAreCreated();
+			_persister.Insert(data, _correlations.Keys.ToArray());
+
+			Assert.Multiple(() =>
+			{
+				foreach (var correlation in _correlations)
+				{
+					var value = correlation.Value.GetValue(data);
```

Comment:
This was probable better served by using _correlations as a TestCaseSource.. thus having many individual tests, one for each data type. But this will imply storing CompleSaga each time.. so it is not w/o tradeoffs..

Reply:
This was more a comment / open idea than a change request.. As I said, there are tradeoffs, and I was looking into your own vision (or maybe solutions for such tradeoffs). Being as it is.. let it stay as is.